scalafix-tests/unit/src/test/scala/scalafix/tests/interfaces/ScalafixSuite.scala

```diff
@@ -0,0 +1,50 @@
+package scalafix.tests.interfaces
+
+import java.net.URL
+
+import coursierapi.Repository
+import org.scalatest.FunSuite
+import scalafix.interfaces.Scalafix
+
+import scala.jdk.CollectionConverters._
+
+/**
+ * Tests in this suite require scalafix-cli & its dependencies to be published (`sbt +publishLocal`) so that
+ * Coursier can fetch them.
```

Comment:
This is annoying, but I really wanted to exercise different Scala binary versions between the client & the implementation to confirm the proper isolation. Initially, I had every scala version (as set in sbt) run against every scala version (as previously published, fetched via Coursier), but I couldn't find a nice way to automate the `publishLocal` step across versions, as it cannot be done in a task. https://github.com/sbt/sbt-projectmatrix could help with that, but would require big changes in the build.

I ended up hardcoding a single version as the target no matter what the sbt version is, and picking the default sbt version so when running that default sbt version (2.13.2), https://github.com/scalacenter/scalafix/pull/1152/files#diff-fdc3abdfd754eeb24090dbd90aeec2ceR246 makes the step unecessary.

Reply:
The metals build contains custom sbt logic to publish local across different Scala versions if you want to take a second stab at this

Reply:
Great, thanks for the pointer, I'll look at it - maybe I'll follow-up in a separate PR though.

Reply:
@olafurpg I managed to adapt https://github.com/scalameta/metals/commit/3ab0608f4660793e25cff324d475fd334dadc7bb (which is just gold!), to do exactly what I wanted (modulo [a scala bug](https://github.com/scala/bug/issues/10603) I ran into) https://github.com/scalacenter/scalafix/compare/4a382e5f..a082cd3e :tada:

The overhead of cross-publishing all artifacts before running tests is surprisingly acceptable, so I think this is ready to go.